eahub/tests/test_profile_models.py

```diff
@@ -57,3 +52,95 @@ def test_get_exportable_field_names(self):
         ]
 
         self.assertListEqual(expected_field_names, actual)
+
+    def test_save_analytics_on_profile_creation(self):
+        profile = create_profile("test@email.com", "User1")
+
+        analytics_logs = ProfileAnalyticsLog.objects.filter(profile=profile)
+
+        analytics_logs_name = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="name"
+        )
+        analytics_logs_is_approved = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="is_approved"
+        )
+        analytics_logs_is_public = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="is_public"
+        )
+        analytics_logs_slug_changed = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="slug_changed"
+        )
+        analytics_logs_slug = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="slug"
+        )
+        analytics_logs_id = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="id"
+        )
+        analytics_logs_user_id = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="user_id"
+        )
+        analytics_logs_email_visible = ProfileAnalyticsLog.objects.filter(
+            profile=profile, field="email_visible"
+        )
+
+        self.assertEqual("User1", analytics_logs_name[0].value)
+        self.assertEqual("False", analytics_logs_is_approved[0].value)
```

Comment:
FYI you can do `ProfileAnalyticsLog.objects.filter(profile=profile, field="is_approved").first() :)